scripts/main.py

```diff
@@ -1,11 +1,21 @@
+import argparse
+
+import nisapi
 import yaml
+
 import iup
-import argparse
 
 
-def run(config):
+def run(config: dict, cache: str):
+    data = nisapi.get_nis(cache)
+
     # List of the cumulative data sets described in the yaml
-    cumulative_data = [iup.parse_nis(**x) for x in config["data"].values()]
+    # note that df.filter(**{"column1": value1, "column2": value2}) is equivalent to
+    # df.filter(pl.col("column1")==value1, pl.col("column2")==value2)
+    cumulative_data = [
+        iup.CumulativeUptakeData(data.filter(**x["filters"]).collect())
+        for x in config["data"].values()
+    ]
```

Comment:
This performs the filtering of rows, but unnecessary columns are no longer dropped (this had been done by `parse_nis` and `select_columns` which are now gone). For example, "indicator_type" and "indicator" will tag along as columns in the uptake data, even though they always have the same value for every row. I assume this is by design, but just wanted to note it in case you didn't want that.

Reply:
confusingly, the different NIS data sets refer to "got a vaccination" in slightly different ways, with different indicators

so I think this will be some work in whatever script does the pipeline, to figure out exactly what kind of data should get passed in

what's the right place to start here? Covid during 2023/2024?

Reply:
2023/2024 covid sounds like a fine starting point. As long as each data set is filtered to only the "got a vaccination" rows (regardless of how it is worded), then the "indicator_type" and "indicator" columns can be dropped altogether. Which can be done in the script.

Reply:
I started a `scripts/preprocess.py` script and added it to the Makefile. We'll probably want to reconfigure how that reads from the config, etc. But it's a starting point, and the scope of this PR is just to get us started with pulling data from the API, rather than a specific file on disk.